pkg/coscheduling/coscheduling.go

```diff
@@ -370,10 +387,42 @@ func responsibleForPod(pod *v1.Pod) bool {
 func (cs *Coscheduling) podGroupInfoGC() {
 	cs.podGroupInfos.Range(func(key, value interface{}) bool {
 		pgInfo := value.(*PodGroupInfo)
-		if pgInfo.deletionTimestamp != nil && pgInfo.deletionTimestamp.Add(PodGroupExpirationTime).Before(cs.clock.Now()) {
+		if pgInfo.deletionTimestamp != nil && pgInfo.deletionTimestamp.Add(cs.args.PodGroupExpirationTime).Before(cs.clock.Now()) {
 			klog.V(3).Infof("%v is out of date and has been deleted in PodGroup GC", key)
 			cs.podGroupInfos.Delete(key)
 		}
 		return true
 	})
 }
+
+// parsePluginArgs parses optional parameters for Coscheduling plugin.
+func parsePluginArgs(params []byte, args *Args) error {
```

Comment:
Can we leverage the upstream's utility:

https://github.com/kubernetes/kubernetes/blob/v1.18.0/pkg/scheduler/framework/v1alpha1/registry.go#L31

So only defaulting logic needs to be added.

Reply:
> Can we leverage the upstream's utility:
> 
> https://github.com/kubernetes/kubernetes/blob/v1.18.0/pkg/scheduler/framework/v1alpha1/registry.go#L31
> 
> So only defaulting logic needs to be added.

I replaced "Unmarshal" with "framework.DecodeInto", but I don't think it can unmarshal json into time.Duration directly. So I still need to parse the string into time.Duration.

Reply:
We'd better not use Duration as the type, see my comments below.